### DISCUSSION TITLE
Improving PaDiM results via "bad" class

### DISCUSSION THREAD

**Discussion by pirnerjonas** [open-edge-platform/anomalib#1462] [Q&A] [answered]
Hi all,

I am currently working on an anomaly detection task where we use PaDiM with `task=classification`. We already have a working model and with its help we identified some more defect classes (e.g. small scratches). 

1. If we now **add these error cases to the “bad” folder** (and the images also end up in the validation dataset) **can we expect a new model trained on this dataset to be better at detecting small scratches** (because actually a lot of these small scratches didn't get detected by the first model)? 
2. Or do we just additionally validate our model against these new error cases? If so are there other model architectures in anomalib that incorporate "bad" images in their anomaly map?

**Comment by abc-125** [comment, marked as answer]
Hi,
As far as I know, all currently implemented models in anomalib support only one-class anomaly detection (no bad photos for training). If you add bad images to the "bad" folder, they will be used for validation and testing only. 
You might try to use tiling to detect smaller defects.

**Reply by pirnerjonas** [reply]
I see thanks for clearing this up. And if I switch from classification to segmentation task and therefore provide masks of the defects it is also just helping with validating the results (e.g pixel wise metrics), right?

**Reply by abc-125** [reply]
You're welcome! Yes, exactly this.

**Reply by pirnerjonas** [reply]
Okay so if my model is insensitive to some of the new found error cases (we identified them by letting the model predict on new unlabelled data and inspecting the results close to the decision boundary and these new error cases were false negatives) the concrete options I see to improve results of a given model are:

	- Making sure the error cases are not part of the “good” images used for training
	- Tiling if the defects are small

Are there any other things I could try in this case?

**Reply by blaz-r** [reply]
The two options you listed are both valid. Especially the first one, even though Padim is not that sensitive to noise data, it's a good idea to have really only good samples for training.
Tiling could help, but with basic tiling, it might not be that significant. Another option, similar to tiling, is to increase resolution, but that comes at the cost of higher memory usage. You could also train a separate model for each location to reduce memory usage, like in tiled ensemble.
Another option might be to also try some other models.

**Reply by pirnerjonas** [reply]
Hi @blaz-r, thanks for the answer
> You could also train a separate model for each location to reduce memory usage, like in tiled ensemble.

I did not mention it but we actualy already do that.

> Another option, similar to tiling, is to increase resolution, but that comes at the cost of higher memory usage. 

Do you mean by that if we e.g. currently create tiles of size 256x256 we should try 512x512

**Reply by blaz-r** [reply]
Yea, you could try 512x512 if you can. Many models benefit from higher resolution when detecting small anomalies.

**Reply by pirnerjonas** [reply]
thanks for answering my questions @blaz-r and @abc-125 👍 I might create a new discussion if I have follow up questions to the suggestions you gave because my inital question is answered!
